include/cbor/pod.h

```diff
@@ -723,9 +723,9 @@ static result deserializeSignedInteger(Type& v, Data& data)
   std::uint64_t res;
   result advanced = deserializeItem(first, res, data);
   std::uint8_t read_major_type = first >> 5;
-  if (read_major_type == 0b000)
+  if (read_major_type == 0b000)  // positive integer
   {
-    if (res > std::numeric_limits<Type>::max())
+    if (res > static_cast<std::uint64_t>(std::numeric_limits<Type>::max()))
```

Comment:
This was a signed vs unsigned comparison. Since `res` is always a positive integer we can always cast the `Type` into which we are deserializing into a `std::uint64_t`, this also works if `Type` is a signed integer since they always have a lower max then unsigned.